src/seedu/addressbook/Main.java

```diff
@@ -107,6 +108,9 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+            if(!storage.isFileExist()){
+            	throw new StorageFileNotFoundException();
+            }
```

Comment:
This method is a good place to catch the exception, not throw it :-) The missing file is better detected in the save method itself.